### ISSUE TITLE
add random function to natives

### ISSUE THREAD

**Issue by jacoobes** [jacoobes/ScratchPaper#12]

**Comment by HarisSiliris** [comment]
Hello, I would like to take a look at the issue!

**Comment by jacoobes** [comment]
let me know if you have any concerns.

**Comment by HarisSiliris** [comment]
Hello i've dropped a little glance at the code and my main concern was that
i get a "Cannot resolve symbol 'ExprParser'" and in general it seems that
parts of the code that are linked with the expressions mainly are not
resolved, bukkit also.I' ve tried to find the solution by myself but not
had enough time unfortunately, im looking forward for your message if you
can enlighten me. Thank you in advance.

Στις Τετ 30 Οκτ 2024 στις 11:26 μ.μ., ο/η Jacob Nguyen <
***@***.***> έγραψε:

**Comment by jacoobes** [comment]
Hello, you will need to generate that code. there should be a gradle script which allows for this.

Try finding the generateGrammarSource script.

**Comment by jacoobes** [comment]
Here it is for reference:
https://docs.gradle.org/current/userguide/antlr_plugin.html

**Comment by HarisSiliris** [comment]
Thank you, i'll reach out if i need anything else.

**Comment by HarisSiliris** [comment]
Hello i've managed to run the gradle task, but still bukkit is not resloved while it is included in the dependencies via papermc. Do I have to manually download Bukkit or Paper JARs?

**Comment by jacoobes** [comment]
are you running this in intellij? It should have automatically installed dependencies to run the project.

Try running buildDependencies script.

**Comment by HarisSiliris** [comment]
Yes i run it on IntelliJ, I tried to rebuild the dependencies too but again there seems to be a compilation error.Maybe I have to download the paperMC server? The command I ran in terminal was : ./gradlew build --refresh-dependencies

Here is the error message too, for reference: 
Execution failed for task ':compileJava'.
> Could not resolve all files for configuration ':compileClasspath'.
   > Could not resolve io.papermc.paper:paper-api:1.20.5-R0.1-SNAPSHOT.
     Required by:
         project :
      > Could not resolve io.papermc.paper:paper-api:1.20.5-R0.1-SNAPSHOT.
         > Could not parse POM https://repo.papermc.io/repository/maven-public/io/papermc/paper/paper-api/1.20.5-R0.1-SNAPSHOT/paper-api-1.20.5-R0.1-20240429.035133-20.pom
            > Could not find net.kyori:adventure-bom:4.17.0-SNAPSHOT.
              Searched in the following locations:
                - https://repo.maven.apache.org/maven2/net/kyori/adventure-bom/4.17.0-SNAPSHOT/maven-metadata.xml
                - https://repo.maven.apache.org/maven2/net/kyori/adventure-bom/4.17.0-SNAPSHOT/adventure-bom-4.17.0-SNAPSHOT.pom

**Comment by jacoobes** [comment]
Try a different version of paper instead. I am unsure of which ones are working.

https://docs.papermc.io/paper/dev/project-setup

**Comment by HarisSiliris** [comment]
Okk ill try to figure it out.I'll contact back if I have serious trouble.Also for the random function is it a good idea to limit the return results by request of the user by adding a parameter? As an example rand(10) will return random numbers from 0 to 10 rand(100) from 0 to 100 and rand() will have no limit for the expected random number.

**Comment by jacoobes** [comment]
> Okk ill try to figure it out.I'll contact back if I have serious trouble.Also for the random function is it a good idea to limit the return results by request of the user by adding a parameter? As an example rand(10) will return random numbers from 0 to 10 rand(100) from 0 to 100 and rand() will have no limit for the expected random number.

hey, sorry was busy this weekend. a range of numbers would be good so 

rand(0, 10)  // random number from 0 to 10  floating point numbers

**Comment by HarisSiliris** [comment]
Hello im still having difficulties with paperMC.I tries to replace compileOnly 'io.papermc.paper:paper-api:1.20.5-R0.1-SNAPSHOT' in the dependencies with the latest version that is recommended in the link you sent but i still get error messages while im trying to build it. Also the methods in natives that already exist are basically taken from Java Math? Im asking because Java Math already provides a random function but without a manageable range numbers, it is from 0.0 to 1.0 i think. Also i never downloaded paper is it necessary to do it and thats why im facing problems with building the project?

**Comment by jacoobes** [comment]
I will try to reproduce your issues. Thank you for your patience.

**Comment by jacoobes** [comment]
Will apply a fix soon.

**Comment by HarisSiliris** [comment]
Ok my friend, keep me updated when I'll be ready to continue with the issue.

**Comment by jacoobes** [comment]
To continue developing, youll need to sync your fork again. In addition, It seems like the issue you were having was due to .net.kyori.adventure-bom-SNAPSHOT (4.17) https://github.com/KyoriPowered/adventure  being deleted, and 1.20.5 support broken. I updated paper in main and also shadow jar doesnt support version 17 or higher, so i switched to a maintained fork of shadow. 

To switch your jdk, go to the hamburger icon, click file > project structure, and switch to those jars / jdk.
![image](https://github.com/user-attachments/assets/480f0e15-f07e-45f3-b2b5-2bc3f48a44e4)

![image](https://github.com/user-attachments/assets/a7c40eda-c4c8-40bc-b607-e88a884de0a4)
then, gradle clean and gradle build.
